### PR TITLE
[Docs] [Clusters] Fix issues in the overview part of Cluster Deployment Guide, and fix a typo

### PR DESCRIPTION
**Pull request by zhe-thoughts**
## Why are these changes needed?
On the "Cluster Deployment Guide" [page](https://docs.ray.io/en/master/cluster/guide.html), in the initial overview
part, we are missing a section, and the wording can be improved. There was also a typo in "Downscaling"

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :( (Docs-only change)

### CODE REVIEW

**Review by wuisawesome — APPROVED** [review]
Coup over this portion of the docs complete

**Comment by wuisawesome** [comment]
Rebased this PR, the linkcheck failures look unrelated, but a little paranoid since it isn't on the flakey test tracker?

File: doc/source/cluster/guide.rst [hunk 1]
```diff
@@ -9,8 +9,9 @@ This page provides an overview of how to deploy a multi-node Ray cluster, includ
 
 * Launch the cluster.
 * Set up the autoscaler.
+* Deploy a Ray application
```
Comment (DmitriGekhtman) [inline doc/source/cluster/guide.rst:12]: nit: Add a period for consistency with the rest of the bullets.

**Review by DmitriGekhtman — DISMISSED** [review]
Single-character nit.

**Review by zhe-thoughts — COMMENTED** [review]
Addressing @DmitriGekhtman 's comment

**Review by DmitriGekhtman — APPROVED** [review]

**Comment by DmitriGekhtman** [comment]
> Rebased this PR, the linkcheck failures look unrelated, but a little paranoid since it isn't on the flakey test tracker?

I've heard much talk of linkcheck failures. I have no idea what that means, but I think it's a known issue, cc @maxpumperla.

**Pull request merged**